tests/integration/plugins/helpers.py

```diff
@@ -82,9 +82,12 @@ def generate_bulk_training_data(
     result_list = []
     for i in range(docs_count):
         result += json.dumps({"index": {"_index": index_name, "_id": i}}) + "\n"
+        # Breaks the single-dimension (ndim*docs_count samples) data vector above into
+        # a matrix with (ndim, docs_count), where each vector has len() == ndim
         result_list.append([float(data[j]) for j in range(i * dimensions, (i + 1) * dimensions)])
         inter = {vector_name: result_list[i]}
         if has_result:
+            # Naming "result", 1xdocs_count dimension vector, as "price"
             inter["price"] = float(responses[i])
```

Comment:
It is not in the PR, but
just curious why there is a cast to float right before call to json.dums?

Reply:
These should be float values. I am casting it to float to ensure I have "0" and not "0.0"

Reply:
so you need floats everywhere, except 0?